src/box/vinyl.c

```diff
@@ -3029,6 +3029,42 @@ struct vy_join_ctx {
 	struct vy_read_view *rv;
 };
 
+#if defined(ENABLE_FETCH_SNAPSHOT_CHECKPOINT_CURSOR)
+#include "vinyl_checkpoint_join.c"
+#else /* !defined(ENABLE_FETCH_SNAPSHOT_CHECKPOINT_CURSOR) */
+
+static int
+vinyl_engine_prepare_checkpoint_join(struct engine *engine,
+				     struct engine_join_ctx *ctx)
+{
+	(void)engine;
+	(void)ctx;
+	diag_set(ClientError, ER_UNSUPPORTED, "Tarantool CE",
+		 "checkpoint join");
```

Comment:
Nit: This function is unreachable because `engine_prepare_join` fails on memtx.

Reply:
Fixed